internal/auth/auth.go

```diff
@@ -0,0 +1,344 @@
+package auth
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+
+	"backend/internal/models"
+
+	"backend/internal/config"
+	"backend/internal/handlers"
+	"backend/internal/middleware"
+
+	"github.com/gin-contrib/sessions"
+	"github.com/gin-gonic/gin"
+	"github.com/markbates/goth"
+	"github.com/markbates/goth/providers/google"
+	"gorm.io/gorm"
+)
+
+// Add this line to ensure AuthHandler implements Handler interface
+var _ handlers.Handler = (*AuthHandler)(nil)
+
+func InitAuth() error {
+	clientID := os.Getenv("GOOGLE_CLIENT_ID")
+	clientSecret := os.Getenv("GOOGLE_CLIENT_SECRET")
+
+	fmt.Printf("InitAuth - Client ID length: %d\n", len(clientID))
+	fmt.Printf("InitAuth - Client Secret length: %d\n", len(clientSecret))
+
+	if clientID == "" || clientSecret == "" {
+		return fmt.Errorf("google oauth credentials not configured")
+	}
+	cfg, err := config.LoadConfig()
```

Comment:
don't we have a cfg object somewhere that can be passed to this? if not, then it's okay to reinitialize it here. want to avoid having too many initializations of this object.